### PR TITLE
Bugfix for Kisada and switching control of him

### LINKED ISSUE #607: Hida Kisada: First Action initiated is not player specific
If Hida Kisada would change control after recording an opponent's action, it will use that action as a reference whether to cancel or not. Even when the the now "new" opponent (original controller of Kisada) hasn't used an action yet.

### PR DESCRIPTION
**Pull request by The-Outcast**
closes #607

### CODE REVIEW

File: server/game/cards/14.5-AHD/FearlessSkirmisher.js [hunk 1]
```diff
@@ -13,8 +13,8 @@ class FearlessSkirmisher extends DrawCard {
                 token: {
                     activePromptTitle: 'Choose a dishonored token',
                     mode: TargetModes.Token,
-                    tokenCondition: token => {
-                        return token.dishonored;
+                    cardCondition: card => {
+                        return card.dishonored;
```
Comment (marcsoiferman) [inline server/game/cards/14.5-AHD/FearlessSkirmisher.js:17]: Why did you change this?

**Review by marcsoiferman — CHANGES_REQUESTED** [review]

File: server/game/cards/01-Core/HidaKisada.js [hunk 2]
```diff
@@ -4,7 +4,8 @@ const { Locations, CardTypes, EventNames, AbilityTypes } = require('../../Consta
 
 class HidaKisada extends DrawCard {
     setupCardAbilities() {
-        this.firstActionEvent = null;
+        this.firstActionEvent = [];
```
Comment (jeremylarner) [inline server/game/cards/01-Core/HidaKisada.js:7]: This should be `this.firstActionEvent = {};`, no?

File: server/game/cards/01-Core/HidaKisada.js [hunk 3]
```diff
@@ -18,13 +19,17 @@ class HidaKisada extends DrawCard {
     }
 
     onInitiateAbilityEffectsWouldInterrupt(event) {
-        if(!this.firstActionEvent && this.game.isDuringConflict() && event.context.ability.abilityType === 'action' && !event.context.ability.cannotBeCancelled && event.context.player !== this.controller) {
-            this.firstActionEvent = event;
+        if(!this.firstActionEvent) {
```
Comment (jeremylarner) [inline server/game/cards/01-Core/HidaKisada.js:22]: This shouldn't be here

File: server/game/cards/01-Core/HidaKisada.js [hunk 4]
```diff
@@ -18,13 +19,17 @@ class HidaKisada extends DrawCard {
...
+            this.firstActionEvent = [];
+        }
+
+        if(!this.firstActionEvent[event.context.player.id] && this.game.isDuringConflict() && event.context.ability.abilityType === 'action' && !event.context.ability.cannotBeCancelled && event.context.player !== this.controller) {
+            this.firstActionEvent[event.context.player.id] = event;
```
Comment (jeremylarner) [inline server/game/cards/01-Core/HidaKisada.js:27]: This should use `uuid` instead of `id` I think

File: server/game/cards/01-Core/HidaKisada.js [hunk 5]
```diff
@@ -18,13 +19,17 @@ class HidaKisada extends DrawCard {
...
         }
     }
 
     onInitiateAbilityEffectsOtherEffects(event) {
-        if(event === this.firstActionEvent && !event.cancelled && this.location === Locations.PlayArea && !this.isBlank() && !this.game.conflictRecord.some(conflict => conflict.winner === this.controller.opponent)) {
+        if(event === this.firstActionEvent[event.context.player.id] && !event.cancelled && this.location === Locations.PlayArea && !this.isBlank() && !this.game.conflictRecord.some(conflict => conflict.winner === this.controller.opponent)) {
             event.cancel();
             this.game.addMessage('{0} attempts to initiate {1}{2}, but {3} cancels it', event.context.player, event.card, event.card.type === CardTypes.Event ? '' : '\'s ability', this);
         }
```
Comment (jeremylarner) [inline server/game/cards/01-Core/HidaKisada.js:35]: onConflictDeclared() should be changed too

**Review by jeremylarner — CHANGES_REQUESTED** [review]

File: server/game/cards/14.5-AHD/FearlessSkirmisher.js (hunk 1 above)
Comment (The-Outcast) [reply server/game/cards/14.5-AHD/FearlessSkirmisher.js:17]: because this was a hack anyway, just forget to change the other part. is fixed now

File: server/game/cards/01-Core/HidaKisada.js (hunk 2 above)
Comment (The-Outcast) [reply server/game/cards/01-Core/HidaKisada.js]: luckily JS doesn't really care when arrays have empty slots, but yes I'll change it to an object

File: server/game/cards/01-Core/HidaKisada.js (hunk 3 above)
Comment (jeremylarner) [reply server/game/cards/01-Core/HidaKisada.js:22]: This still shouldn't be here....

Comment (The-Outcast) [reply server/game/cards/01-Core/HidaKisada.js]: yeah you are right, problem was the unchanged onConflictDeclared code.

**Review by marcsoiferman — APPROVED** [review]
